### PR TITLE
Adding event types reference page to docs

### PR DESCRIPTION
**Pull request by flossypurse**
Preview: https://deploy-preview-296--mystifying-fermi-1bc096.netlify.app/docs/server-event-types

### CODE REVIEW

File: docs/server-event-types.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+sidebar_label: Event types
+---
+
+Event types are defined in the Server API [event_type.proto](https://github.com/temporalio/api/blob/master/temporal/api/enums/v1/event_type.proto) file.
+
+- **EVENT_TYPE_UNSPECIFIED**:
+- **EVENT_TYPE_WORKFLOW_EXECUTION_STARTED**:
+- **EVENT_TYPE_WORKFLOW_EXECUTION_COMPLETED**:
+- **EVENT_TYPE_WORKFLOW_EXECUTION_FAILED**:
+- **EVENT_TYPE_WORKFLOW_EXECUTION_TIMED_OUT**:
```
Comment (wxing1292) [inline docs/server-event-types.md:13]: - **EVENT_TYPE_UNSPECIFIED**:
This event type is a place holder, will never appear in user's workflow execution history

- **EVENT_TYPE_WORKFLOW_EXECUTION_STARTED**:
This event type indicates the workflow execution is started / triggered.
This event type contains workflow execution inputs, as well as workflow timeout configurations.

- **EVENT_TYPE_WORKFLOW_EXECUTION_COMPLETED**:
This event type indicates the workflow execution is successfully completed.
This event type contains workflow execution results.

- **EVENT_TYPE_WORKFLOW_EXECUTION_FAILED**:
This event type indicates the workflow execution is unsuccessfully completed.
This event type contains workflow execution error.

- **EVENT_TYPE_WORKFLOW_EXECUTION_TIMED_OUT**:
This event type indicates the workflow execution is timed out by Temporal server, due to workflow not completed within timeout settings.

- **EVENT_TYPE_WORKFLOW_EXECUTION_TERMINATED**:
This event type indicates the workflow execution is forcefully terminated, e.g. caller call terminate workflow API on the Temporal server side. 

**EVENT_TYPE_WORKFLOW_EXECUTION_CANCELED**:
This event type indicates the workflow execution is cancelled, e.g. caller call cancel workflow API on the Temporal server side, Temporal server let SDK know workflow should be cancelled, SDK decided to cancel workflow.

- **EVENT_TYPE_WORKFLOW_EXECUTION_CONTINUED_AS_NEW**:
This event type indicates workflow is successfully completed, and a new workflow should be started within the same transaction.
This event type contains last workflow execution results, as well as new workflow execution inputs.

File: docs/server-event-types.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+- **EVENT_TYPE_WORKFLOW_TASK_SCHEDULED**:
+- **EVENT_TYPE_WORKFLOW_TASK_STARTED**:
+- **EVENT_TYPE_WORKFLOW_TASK_COMPLETED**:
+- **EVENT_TYPE_WORKFLOW_TASK_TIMED_OUT**:
+- **EVENT_TYPE_WORKFLOW_TASK_FAILED**:
```
Comment (wxing1292) [inline docs/server-event-types.md:18]: - **EVENT_TYPE_WORKFLOW_TASK_SCHEDULED**:
This event type indicates the workflow task scheduled, meaning SDK should process the new history events

- **EVENT_TYPE_WORKFLOW_TASK_STARTED**:
This event type indicates the workflow task started, meaning SDK picked up the workflow task, and is processing new history events

- **EVENT_TYPE_WORKFLOW_TASK_COMPLETED**:
This event type indicates the workflow task completed, meaning SDK picked up the workflow task, processed new history events, and may / may not ask Temporal server to do additional work, see following events:
  - **EVENT_TYPE_ACTIVITY_TASK_SCHEDULED**
  - **EVENT_TYPE_TIMER_STARTED**
  - **EVENT_TYPE_UPSERT_WORKFLOW_SEARCH_ATTRIBUTES**
  - **EVENT_TYPE_MARKER_RECORDED**
  - **EVENT_TYPE_START_CHILD_WORKFLOW_EXECUTION_INITIATED**
  - **EVENT_TYPE_REQUEST_CANCEL_EXTERNAL_WORKFLOW_EXECUTION_INITIATED**
  - **EVENT_TYPE_SIGNAL_EXTERNAL_WORKFLOW_EXECUTION_INITIATED**
  - **EVENT_TYPE_WORKFLOW_EXECUTION_COMPLETED**
  - **EVENT_TYPE_WORKFLOW_EXECUTION_FAILED**
  - **EVENT_TYPE_WORKFLOW_EXECUTION_CANCELED**
  - **EVENT_TYPE_WORKFLOW_EXECUTION_CONTINUED_AS_NEW**

- **EVENT_TYPE_WORKFLOW_TASK_TIMED_OUT**:
This event type indicates the workflow task encountered timeout, meaning either a SDK with local cache is not available at the time, or it took too long for a SDK to process this task.

- **EVENT_TYPE_WORKFLOW_TASK_FAILED**:
This event type indicates the workflow task encountered failure, usually meaning non deterministic workflow. 
NOTE: workflow reset functionality also use this event.

File: docs/server-event-types.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+- **EVENT_TYPE_ACTIVITY_TASK_SCHEDULED**:
+- **EVENT_TYPE_ACTIVITY_TASK_STARTED**:
+- **EVENT_TYPE_ACTIVITY_TASK_COMPLETED**:
+- **EVENT_TYPE_ACTIVITY_TASK_FAILED**:
+- **EVENT_TYPE_ACTIVITY_TASK_TIMED_OUT**:
+- **EVENT_TYPE_ACTIVITY_TASK_CANCEL_REQUESTED**:
+- **EVENT_TYPE_ACTIVITY_TASK_CANCELED**:
```
Comment (wxing1292) [inline docs/server-event-types.md:25]: - **EVENT_TYPE_ACTIVITY_TASK_SCHEDULED**:
This event type indicates the activity task scheduled, meaning SDK should pick up this activity task and execute.
This event type contains activity inputs, as well as activity timeout configurations.

- **EVENT_TYPE_ACTIVITY_TASK_STARTED**:
This event type indicates the activity task started, meaning SDK picked up the activity task, and is processing this activity invocation.

- **EVENT_TYPE_ACTIVITY_TASK_COMPLETED**:
This event type indicates the activity task completed, meaning SDK picked up the activity task, and successfully completed the activity task.
This event type contains activity execution result.

- **EVENT_TYPE_ACTIVITY_TASK_FAILED**:
This event type indicates the activity task completed, meaning SDK picked up the activity task, and unsuccessfully completed the activity task.
This event type contains activity execution error.

- **EVENT_TYPE_ACTIVITY_TASK_TIMED_OUT**:
This event type indicates the activity is timed out by Temporal server, due to activity not completed within timeout settings.

- **EVENT_TYPE_ACTIVITY_TASK_CANCEL_REQUESTED**:
This event type indicates the activity should be cancelled. SDK will know an activity should be cancelled during activity heartbeat. 

- **EVENT_TYPE_ACTIVITY_TASK_CANCELED**:
This event type indicates the activity is cancelled.

File: docs/server-event-types.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+- **EVENT_TYPE_TIMER_STARTED**:
+- **EVENT_TYPE_TIMER_FIRED**:
+- **EVENT_TYPE_TIMER_CANCELED**:
```
Comment (wxing1292) [inline docs/server-event-types.md:28]: - **EVENT_TYPE_TIMER_STARTED**:
This event type indicates the timer has started.

- **EVENT_TYPE_TIMER_FIRED**:
This event type indicates the timer has fired.

- **EVENT_TYPE_TIMER_CANCELED**:
This event type indicates the timer has been cancelled.

File: docs/server-event-types.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+- **EVENT_TYPE_WORKFLOW_EXECUTION_CANCEL_REQUESTED**:
```
Comment (wxing1292) [inline docs/server-event-types.md:29]: - **EVENT_TYPE_WORKFLOW_EXECUTION_CANCEL_REQUESTED**:
This event type indicates the workflow should be cancelled.

File: docs/server-event-types.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+- **EVENT_TYPE_WORKFLOW_EXECUTION_CANCELED**:
+- **EVENT_TYPE_REQUEST_CANCEL_EXTERNAL_WORKFLOW_EXECUTION_INITIATED**:
+- **EVENT_TYPE_REQUEST_CANCEL_EXTERNAL_WORKFLOW_EXECUTION_FAILED**:
+- **EVENT_TYPE_EXTERNAL_WORKFLOW_EXECUTION_CANCEL_REQUESTED**:
```
Comment (wxing1292) [inline docs/server-event-types.md:33]: - **EVENT_TYPE_REQUEST_CANCEL_EXTERNAL_WORKFLOW_EXECUTION_INITIATED**:
This event type indicates that Temporal server will try to cancel the target workflow.

- **EVENT_TYPE_REQUEST_CANCEL_EXTERNAL_WORKFLOW_EXECUTION_FAILED**:
This event type indicates that Temporal server cannot cancel the target workflow, usually due to target workflow not found.

- **EVENT_TYPE_EXTERNAL_WORKFLOW_EXECUTION_CANCEL_REQUESTED**:
This event type indicates that Temporal server successfully request the target workflow to be cancelled.

File: docs/server-event-types.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+- **EVENT_TYPE_MARKER_RECORDED**:
+- **EVENT_TYPE_WORKFLOW_EXECUTION_SIGNALED**:
```
Comment (wxing1292) [inline docs/server-event-types.md:35]: - **EVENT_TYPE_WORKFLOW_EXECUTION_SIGNALED**:
This event type indicates the workflow has received a signal event.
This event type contains signal name, as well as signal payload.

File: docs/server-event-types.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+- **EVENT_TYPE_WORKFLOW_EXECUTION_TERMINATED**:
+- **EVENT_TYPE_WORKFLOW_EXECUTION_CONTINUED_AS_NEW**:
+- **EVENT_TYPE_START_CHILD_WORKFLOW_EXECUTION_INITIATED**:
+- **EVENT_TYPE_START_CHILD_WORKFLOW_EXECUTION_FAILED**:
+- **EVENT_TYPE_CHILD_WORKFLOW_EXECUTION_STARTED**:
+- **EVENT_TYPE_CHILD_WORKFLOW_EXECUTION_COMPLETED**:
+- **EVENT_TYPE_CHILD_WORKFLOW_EXECUTION_FAILED**:
+- **EVENT_TYPE_CHILD_WORKFLOW_EXECUTION_CANCELED**:
+- **EVENT_TYPE_CHILD_WORKFLOW_EXECUTION_TIMED_OUT**:
+- **EVENT_TYPE_CHILD_WORKFLOW_EXECUTION_TERMINATED**:
```
Comment (wxing1292) [inline docs/server-event-types.md:45]: - **EVENT_TYPE_START_CHILD_WORKFLOW_EXECUTION_INITIATED**:
This event type indicates that Temporal server will try to start a child workflow.

- **EVENT_TYPE_START_CHILD_WORKFLOW_EXECUTION_FAILED**:
This event type indicates the child workflow execution cannot be started / triggered, usually due to child workflow ID collision.

- **EVENT_TYPE_CHILD_WORKFLOW_EXECUTION_STARTED**:
This event type indicates the child workflow execution is successfully started / triggered.
Also see **EVENT_TYPE_WORKFLOW_EXECUTION_STARTED**

- **EVENT_TYPE_CHILD_WORKFLOW_EXECUTION_COMPLETED**:
This event type indicates the child workflow execution is successfully completed.
Also see **EVENT_TYPE_WORKFLOW_EXECUTION_COMPLETED**

- **EVENT_TYPE_CHILD_WORKFLOW_EXECUTION_FAILED**:
This event type indicates the child workflow execution is unsuccessfully completed.
Also see **EVENT_TYPE_WORKFLOW_EXECUTION_FAILED**

- **EVENT_TYPE_CHILD_WORKFLOW_EXECUTION_CANCELED**:
This event type indicates the child workflow execution is cancelled. 
Also see **EVENT_TYPE_WORKFLOW_EXECUTION_CANCELED**

- **EVENT_TYPE_CHILD_WORKFLOW_EXECUTION_TIMED_OUT**:
This event type indicates the child workflow execution is timed out by Temporal server. 
Also see **EVENT_TYPE_WORKFLOW_EXECUTION_TIMED_OUT**

- **EVENT_TYPE_CHILD_WORKFLOW_EXECUTION_TERMINATED**:
This event type indicates the child workflow execution terminated. 
Also see **EVENT_TYPE_WORKFLOW_EXECUTION_TERMINATED**

**Review by sw-yx — APPROVED** [review]

**Review by wxing1292 — APPROVED** [review]

**Pull request merged**